src/get-glamor-classname.js

```diff
@@ -10,17 +10,23 @@ import {css, styleSheet} from 'glamor'
  *   - glamorlessClassName is the rest of the className string
  *     without the glamor classNames
  */
-function extractGlamorStyles(className = '') {
-  return className.toString().split(' ').reduce((groups, name) => {
+function extractGlamorStyles(className) {
+  const glamorlessClassName = []
+  const glamorStyles = []
+  className.toString().split(' ').forEach(name => {
     if (name.indexOf('css-') === 0) {
-      const style = getGlamorStylesFromClassName(name)
-      groups.glamorStyles.push(style)
+      const style = buildGlamorSrcFromClassName(name)
+      glamorStyles.push(style)
     } else {
-      // eslint-disable-next-line max-len
-      groups.glamorlessClassName = `${groups.glamorlessClassName} ${name}`.trim()
+      glamorlessClassName.push(name)
     }
-    return groups
-  }, {glamorlessClassName: '', glamorStyles: []})
+  })
+
+  return {glamorlessClassName, glamorStyles}
+}
+
+function buildGlamorSrcFromClassName(className) {
```

Comment:
Could we add a comment above this line giving a brief explanation of how this black magic works?